tests/generation/FileGlobs.py

```diff
@@ -97,6 +98,98 @@ def test_get_project_files_gitignore(self):
         os.remove(".gitignore")
         os.chdir(orig_cwd)
 
+    def test_get_project_files_npmignore(self):
+        orig_cwd = os.getcwd()
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.makedirs("file_globs_npmignore_testfiles", exist_ok=True)
+        os.chdir("file_globs_npmignore_testfiles")
+
+        with open(".npmignore", "w") as f:
+            f.write("""
+# Start of npmignore
+build
+ignore.c
+/tests
+/upload.c
+/*.py
+*.pyc
+__pycache__
+# End of npmignore""")
+
+        os.makedirs("other_folder", exist_ok=True)
+        os.chdir("other_folder")
+        with open('.npmignore', "w") as file:
+            file.write("""
+#Start of npmignore
+*.html
+#End of npmignore""")
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.chdir("file_globs_npmignore_testfiles")
+        os.makedirs("sample_data", exist_ok=True)
+        os.chdir("sample_data")
```

Comment:
There seem to be too many `chdir` operations. Can't this be done with combined use of `os.makedirs` and `open` alone?